### ISSUE TITLE
Admin Page endless waiting after hitting F5

### ISSUE THREAD

**Issue by cslegany-synerinsoft** [strapi/strapi#21811]
### Required System information

- Node.js version: v20.11.0
- NPM version: 10.2.4
- Strapi version: 5.0.6
- Database: SQLite
- Operating system: Windows
- Is your project Javascript or Typescript: Typescript

### Describe the bug
The following bug only happens in production mode when the project is deployed to azure. I was unable to reproduce it locally.
If you press F5 to get the entire site reloaded, sometimes you end up with an infinite loading (the blue loading spinner never ends).
There're no errors in the console.

### Steps to reproduce the behavior

1. Go to the admin page url in Chrome in incognito mode
2. If it loads, press F5 a few times. Sometimes the page loads, sometimes it doesn't and you'll end up with infinite loading.
3. If it doesn't load, press the Back button on the browser bar to get rid of the infinite loading. You can see the error either on the login screen or after getting logged in on the Home screen, or even in Content Manager.
4. See attached screenshot.

### Expected behavior

No infinite loading.

### Screenshots
![F5_timeout](https://github.com/user-attachments/assets/6b7f85db-bc7a-4835-a568-37cdf28205aa)

## package.json
"dependencies": {
    "@_sh/strapi-plugin-ckeditor": "^4.0.6",
    "@strapi/email": "5.0.6",
    "@strapi/plugin-cloud": "5.0.6",
    "@strapi/plugin-users-permissions": "5.0.6",
    "@strapi/strapi": "5.0.6",
    "better-sqlite3": "9.4.3",
    "moment": "^2.30.1",
    "mysql2": "^3.11.3",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-router-dom": "^6.0.0",
    "styled-components": "^6.0.0"
  }
  
  ## config
  All default in middlewares.ts except "frame-src" in "strapi::security" for strapi-plugin-ckeditor.

**Comment by Mastercoder424** [comment]
I can solve it 
assign me 
I have a knowledge of this loader problem

**Comment by tprim92** [comment]
This is a serious bug and needs an immediate fix. It is impossible to use Strapi on our production environment and our customer keeps complaining about it.

**Comment by Convly** [comment]
Hey, thanks for reporting this issue.

Would it be possible to provide more information about what request is preventing the admin from loading?

If you could share screenshots from the network tab or something similar (if possible), that would help us a lot to understand what could be the issue.

**Comment by tprim92** [comment]
Nothing interesting...

![console_tab](https://github.com/user-attachments/assets/4aa75a94-ae9b-4587-8972-f2c6c441b1c8)
![network_tab](https://github.com/user-attachments/assets/a4685fde-995b-4b99-95c7-75a4aeff72d8)

**Comment by Marc-Roig** [comment]
@tprim92 can you also check in the network tab for any blocked request (not filtering by Fext/XHR but showing all requests)

**Comment by tprim92** [comment]
All status responses are 200 OK.
The Strapi website has been deployed to multiple environments. The faster the hosting service, the less frequently occurs the issue.
I'm 100% sure that the issue isn't caused by Azure and I'd believe that an async await completes too fast.

**Comment by Marc-Roig** [comment]
Just in case, could you try running g it locally using yarn build & yarn start ? 🙏

**Comment by Marc-Roig** [comment]
also , I assume thats the case by your words , but it never hurts to ask, are you running this instance in dev mode? (yarn develop) and not production mode (yarn start) ?

**Comment by cslegany-synerinsoft** [comment]
Of course we do develop in dev mode but the Strapi instance has been tested in production mode (npm run build, npm run start) as well. The issue never happens locally, either in dev or production mode.

**Comment by derrickmehaffy** [comment]
> Of course we do develop in dev mode but the Strapi instance has been tested in production mode (npm run build, npm run start) as well. The issue never happens locally, either in dev or production mode.

The reason we ask is that Strapi 5 development mode uses vite which uses a websocket in dev mode. My guess was that in dev mode the port used for that websocket was being blocked or had some kind of race condition.

If you can get it to load at all though that may not be the issue (if you run in prod mode that websocket isn't used)

Thanks for the clarification

**Comment by cslegany-synerinsoft** [comment]
Our guess that this issue is related to speed problems. Strapi runs ultra fast on localhost and much slower on our dev/test environment in Azure. The staging environment is faster and the issue happens less frequently, but it occurs sometimes.

We tried to slow down Strapi in the local environment to reproduce the issue, but in vain. Here are the methods that we've experimented with.
- inject js code like syncWait(5000); into node_modules/.strapi/vite/deps to slow it down using the following code:
`const syncWait = ms => {
    const end = Date.now() + ms
    while (Date.now() < end) continue
}`

Result: we've managed to slow down the login screen or the content manager but the issue couldn't be reproduced.
- use a vpn and the remote DB. 
Result: Strapi was slower but the issue couldn't be reproduced.
- use port forwarding

However we've noticed that port forwarding fails with 429 error. See [here](https://github.com/strapi/strapi/issues/21874).

**Comment by rgellen** [comment]
I have experienced very same issue with Strapi 5, and can confirm that sometimes I end up with an endless timeout. Load errors depend on internet speed and host environment speed.
For example it is impossible to load the admin page when using Strapi with a vpn. It ends up with an endless timeout in all cases, meaning that a slow vpn can cause serious issues.

**Comment by Convly** [comment]
> I have experienced very same issue with Strapi 5, and can confirm that sometimes I end up with an endless timeout. Load errors depend on internet speed and host environment speed. For example it is impossible to load the admin page when using Strapi with a vpn. It ends up with an endless timeout in all cases, meaning that a slow vpn can cause serious issues.

Interesting. Could you share what kind of hosting/VPN combination you're using?

It would help us a lot if we could reproduce the issue on our end.

**Comment by hoqkhanh** [comment]
I have the same issue. Do we have any fixing PR for it?

**Comment by bolg55** [comment]
Internal TID8638 having same issue

**Comment by hoqkhanh** [comment]
I have the same issue when trying to access the CMS after logging in. It loads forever with white page + loader until I clear all browser cache or reload F5 many times.

**Comment by hoqkhanh** [comment]
> > I have experienced very same issue with Strapi 5, and can confirm that sometimes I end up with an endless timeout. Load errors depend on internet speed and host environment speed. For example it is impossible to load the admin page when using Strapi with a vpn. It ends up with an endless timeout in all cases, meaning that a slow vpn can cause serious issues.
> 
> Interesting. Could you share what kind of hosting/VPN combination you're using?
> 
> It would help us a lot if we could reproduce the issue on our end.

I am using the ubuntu server with node 18 and proxy pass of Apache. Not use the VPN

**Comment by huandu** [comment]
I have the same issue with v5.1.0. The admin panel works on my own dev box but keeps loading when deployed.

I spent several days to debug this issue and finally find out the cause and a possible workaround. However, I don't know how to fix it properly. Need help from Strapi dev team. cc @Convly 

TL; DR The `<PluginsInitializer>` becomes `memo` at runtime for unknown reason. The workaround is to make all plugins' `isReady: true` when calling `app.registerPlugin()` in plugin's `admin/src/index.js|ts`.

`<PluginsInitializer>` is designed to run all initializers of plugins. It will render `<Page.Loading />` if some plugins are not ready (the `isReady` flag), and it will render Admin Panel after all plugins are ready. The issue here is that `<PluginsInitializer>` becomes a memo component for unknown reason at runtime (verified by React Dev Tool) so that it will not re-render itself after all plugins are ready.

I don't know why `<PluginsInitializer>` becomes memo. `<PluginsInitializer>` has only one child which is a memo component (the `<DndProvider>`). Maybe React runtime does some magic in this scenario?

**Comment by huandu** [comment]
> The `<PluginsInitializer>` becomes memo at runtime for unknown reason.

I need to correct myself. It's not true. I just read a wrong value. The component type is `$$typeof: Symbol(react.element)`. However, `<PluginsInitializer>` did work like a memo component for unknown reason. The component render function returned correct value, but React didn't refresh component tree.

**Comment by reslear** [comment]
is there a temporary solution?

**Comment by hoqkhanh** [comment]
is there any solution for production mode?

**Comment by reslear** [comment]
I guess I need to downgrade to strapi v4.

**Comment by reslear** [comment]
Temp works is open in new incognito window

**Comment by johnspeedppc** [comment]
> I guess I need to downgrade to strapi v4, because it's impossible to work in production.
> 
> Temp works is open in new incognito window

It works with node_environment = develop

**Comment by CMatias** [comment]
I was able to reproduce this issue locally and in my case, it was the strapi cloud plugin that wasn't ready at the time the initialization was happening. After removing the plugin the issue stopped. Tested in both development and production build. Have not tested in a production deployment.

**Comment by reslear** [comment]
@CMatias you mean @strapi/plugin-cloud ? I don't have it, by the way. By the way, let's share our plugins and see if we can find a pattern. 


```json
{

  // ...

  "dependencies": {
    "@_sh/strapi-plugin-ckeditor": "^4.0.10",
    "@strapi/plugin-seo": "^2.0.2",
    "@strapi/plugin-users-permissions": "5.3.0",
    "@strapi/provider-email-sendgrid": "^5.3.0",
    "@strapi/provider-upload-aws-s3": "^5.3.0",
    "@strapi/strapi": "5.3.0",
    "better-sqlite3": "11.5.0",
    "pg": "^8.13.1",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "react-router-dom": "^6.22.3",
    "strapi-plugin-config-sync": "^3.0.2",
    "strapi-prometheus": "^2.1.0",
    "styled-components": "^6.1.8"
  },
  "devDependencies": {
    "@types/node": "^20.17.6",
    "@types/react": "^18.3.12",
    "@types/react-dom": "^18.3.1",
    "dotenv-cli": "^7.4.2",
    "prettier": "^3.3.3",
    "typescript": "^5.6.3"
  },
  "engines": {
    "node": ">=18.0.0 <=20.x.x",
    "npm": ">=6.0.0"
  },
  "strapi": {
    "telemetryDisabled": true
  }
}

**Comment by hoqkhanh** [comment]
After I removed @strapi/plugin-cloud, still issue

**Comment by hoqkhanh** [comment]
After a month, Do you have any plan to solve it @derrickmehaffy ? I am using the enterprise edition. This issue causes our problems

**Comment by reslear** [comment]
Are the developers aware of this problem? Because I don't even see it in “inprogress”.

**Comment by khanhhood** [comment]
> Are the developers aware of this problem? Because I don't even see it in “inprogress”.

I asked them in the premium support system for the enterprise edition that is known as the strapi team is handling it

**Comment by CMatias** [comment]
@reslear yes that was the plugin, I figured that out by applying a patch on [this file of the admin dashboard](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/components/PluginsInitializer.tsx). As you can see on the file, it has several mentions that it should be improved. I simply logged the plugins every time to understand which were the ones not ready and in my case it was the strapi-cloud plugin (which was ok to remove since I was not using it atm).

My solution was simply a quick fix to move forward, it warrants a deeper dive.

**Comment by reslear** [comment]
@CMatias How to debug a not working plugin, where to place console.log?

**Comment by johnspeedppc** [comment]
> @reslear yes that was the plugin, I figured that out by applying a patch on [this file of the admin dashboard](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/components/PluginsInitializer.tsx). As you can see on the file, it has several mentions that it should be improved. I simply logged the plugins every time to understand which were the ones not ready and in my case it was the strapi-cloud plugin (which was ok to remove since I was not using it atm).
> 
> My solution was simply a quick fix to move forward, it warrants a deeper dive.

Could you give me your patch?

**Comment by CMatias** [comment]
So what I did to figure that out was:

1. Clone the Strapi repo and installed the deps
2. Go into [this file](https://github.com/strapi/strapi/blob/develop/packages/core/admin/admin/src/components/PluginsInitializer.tsx) and added the following on line 29

```
console.log('PluginsInitializer', {
    hasApluginNotReady,
    plugins,
    notReady: Object.keys(plugins).find((plugin) => plugins[plugin].isReady === false),
  });
```
3. I then built the core/admin Strapi package by running `yarn build` in `/packages/core/admin` - this will create a `dist` folder under `/packages/core/admin/dist`
4. Run `pnpm patch @strapi/admin` in my CMS repo and replace the `dist` by the built one (`patch-package` would probably work as well but I didn't test it since I'm using pnpm)
5. Commit the patch
6. Run Strapi and through the logs I've found that the strapi-cloud plugin was the one not being initialised


Just to be clear, this didn't solve anything, it just helped me debug the issue.

**Comment by khanhhood** [comment]
Any update?

**Comment by derrickmehaffy** [comment]
> it caused a big problem with the production. @derrickmehaffy is there any temporary solution for production mode? Or any plan to solve it?
>  I am using a strapi enterprise edition which has the same issue 

I don't have an update that would be up to our engineering team.

For everyone in here though:

- ALL `@strapi/*` packages SHOULD NOT have a `^` in the version 
- Please try deleting your node modules folder and any lock file
- Try removing any optional plugins
- Ensure any package cache at the OS level is cleared (yarn for example)

As far as I'm aware we haven't actually been able to reproduce this cleanly in a fresh project.

**Comment by cosminneaga** [comment]
In my case I removed `@strapi/plugin-cloud` and it started working, infinite loading has vanished on production mode, running on localhost.

**Comment by khanhhood** [comment]
> > it caused a big problem with the production. @derrickmehaffy is there any temporary solution for production mode? Or any plan to solve it?
> > I am using a strapi enterprise edition which has the same issue
> 
> I don't have an update that would be up to our engineering team.
> 
> For everyone in here though:
> 
> * ALL `@strapi/*` packages SHOULD NOT have a `^` in the version
> * Please try deleting your node modules folder and any lock file
> * Try removing any optional plugins
> * Ensure any package cache at the OS level is cleared (yarn for example)
> 
> As far as I'm aware we haven't actually been able to reproduce this cleanly in a fresh project.

I don't think it works because we tried a lot of ways. I think it's an issue from Strapi Core

**Comment by eruiz-impact** [comment]
Is there an ETA on this getting fixed? This is such a major bug that isn't being addressed.

**Comment by derrickmehaffy** [comment]
> Is there an ETA on this getting fixed? This is such a major bug that isn't being addressed. 

It's currently in the engineering backlog at the moment, until one of the squads can get some time to put it in their active queue I don't have any information on when it'll be worked on.

Me and my team only do the initial triage of the issue and assign it to the backlog, our engineering team are the ones who pick it up from there.

**Comment by derrickmehaffy** [comment]
If you want some context you can see the active backlog here: https://github.com/orgs/strapi/projects/34

**Comment by khanhhood** [comment]
Could you assign it to a high priority? For now, It doesn't have priority level @derrickmehaffy

**Comment by khanhhood** [comment]
It's the highest priority issue. May your team should handle it first? @derrickmehaffy

**Comment by andreas-vm** [comment]
I just wanted to share that I'm also experiencing this issue both in production and locally, in development mode and in a local production build running in a Docker Compose setup.

The descriptions provided above accurately reflect what I've encountered. One additional point I can add is that it's very rare in dev mode when using the super admin user (almost impossible to reproduce), but it becomes more common when I use a limited user with a custom role (which is what all the end users will use in our setup).

When I get stuck on the loading spinner, repeated reloads don't seem to work. However, if I wait a bit, return to the tab, and then reload, it seems to load properly.

**Comment by CMatias** [comment]
> The descriptions provided above accurately reflect what I've encountered. One additional point I can add is that it's very rare in dev mode when using the super admin user (almost impossible to reproduce), but it becomes more common when I use a limited user with a custom role (which is what all the end users will use in our setup).

I second this, it also happened during my testing - it was way harder to reproduce the issue with a Super admin.

**Comment by derrickmehaffy** [comment]
Notes that you all, I'll do some extra testing in a prod mode application with a restricted non super admin user

**Comment by brunoxmetal** [comment]
In my case, if I enable the isReady when registering the plugin it works:
![image](https://github.com/user-attachments/assets/04a81cb6-39b4-4d26-bf29-2998fb4d0372)
Bad, because I need Auth before initialize, but the admin panel works.

**Comment by andreas-vm** [comment]
A quick follow up. 

On version 5.4.1 with `isReady: true` for our only custom plugin and where we have removed the documentation plugin completely since last deploy, we are no longer seeing the issue. 

I havn't had time to check what did it (if it was the update, the isReady or the complete removal of documentation plugin).

**Comment by khanhhood** [comment]
> A quick follow up.
> 
> On version 5.4.1 with `isReady: true` for our only custom plugin and where we have removed the documentation plugin completely since last deploy, we are no longer seeing the issue.
> 
> I havn't had time to check what did it (if it was the update, the isReady or the complete removal of documentation plugin).

I don't have a document plugin, but this issue is still happening

**Comment by reslear** [comment]
> > A quick follow up.
> > On version 5.4.1 with `isReady: true` for our only custom plugin and where we have removed the documentation plugin completely since last deploy, we are no longer seeing the issue.
> > I havn't had time to check what did it (if it was the update, the isReady or the complete removal of documentation plugin).
> 
> I don't have a document plugin, but this issue is still happening

also don't have a document plugin

**Comment by pksr** [comment]
I have the same problem and found that if I remove "telemetryDisabled: true" in the package.json it works. Unfortunately I haven't been able to find the problem in the code yet. But I'm not giving up :-).

**Comment by khanhhood** [comment]
2 months passed. But this issue's not solved yet. Do you have any update?

**Comment by reslear** [comment]
> "telemetryDisabled: true"

really works...

**Comment by khanhhood** [comment]
> > "telemetryDisabled: true"
> 
> i tested, really works...

Could you detail what you did? I didn't find "telemetryDisabled" in my packages.json

**Comment by khanhhood** [comment]
> we use telemetryDisabled: false for prevent all url's analyctics because have limited traffic. and after remove ~60% fixed this problem

@reslear Could you share me your packge.json? I don't where to put the telemetryDisabled to?

**Comment by pksr** [comment]
@khanhhood In my case, this is not something you need to set, just remove. So if you don't have this setting in your package.json, then that is probably not the cause of your problem. The other option is that you have the "STRAPI_TELEMETRY_DISABLED" environment variable set. This should have the same effect as the setting in the package.json.

**Comment by pksr** [comment]
I don't know exactly what the problem has to to with the telemetry but I fixed it - hopefully. As far as I could find out, it may be that the process is too fast without telemetry and that states overlap.

Whatever, see PR.

**Comment by Edwin-Luijten** [comment]
For all my custom plugins I added the following property `isReady: true`:
```
app.registerPlugin({
      id: PLUGIN_ID,
      initializer: Initializer,
      isReady: true, <- this resolved the issue
      name: PLUGIN_ID
    });
});
```

This resolved the infinite loading.

**Comment by CodeDruid13** [comment]
@derrickmehaffy I can confirm this loading issue is still in version 5.5.2.

I can create a simple user for you to test and see the issue on the admin panel hosted on digital ocean

**Comment by derrickmehaffy** [comment]
> @derrickmehaffy I can confirm this loading issue is still in version 5.5.2.
> 
> I can create a simple user for you to test and see the issue on the admin panel hosted on digital ocean

We are forbidden from logging into anyone's instance sorry.

**Comment by CodeDruid13** [comment]
> > @derrickmehaffy I can confirm this loading issue is still in version 5.5.2.
> > I can create a simple user for you to test and see the issue on the admin panel hosted on digital ocean
> 
> We are forbidden from logging into anyone's instance sorry.

I fixed, I hope. So far am not seeing the infinite loading

**Comment by khanhhood** [comment]
I've fixed the loading issue on our cms with this patch in 5.3.0
[@strapi+admin+5.3.0.patch](https://github.com/user-attachments/files/18194274/%40strapi%2Badmin%2B5.3.0.patch)

**Comment by Xzandro** [comment]
This definitely seems to have something to do with local plugins maybe or with plugins in general. I have two local plugins that I migrated from v4 to v5. I did not have issues with `npm run develop` at all. I only noticed it after I did a test deploy on a staging env. (npm run build && npm start)

The solution really was, as someone else already said to set the `isReady `to true in the registerPlugin function for both my local plugins. Afterwards I did not see this behavior anymore.
On my other Strapi system without any local plugins, I cannot reproduce this at all.

**Comment by khanhhood** [comment]
Any news for it?

**Comment by cacooper16** [comment]
Yup - experiencing the same issue. 

Only seems to happen on my deployment environment (Digital Ocean), not experienced it on local (I think).

Strapi 4.2

"dependencies": {
    "@strapi/pack-up": "^5.0.2",
    "@strapi/plugin-cloud": "5.2.0",
    "@strapi/plugin-seo": "^2.0.2",
    "@strapi/plugin-users-permissions": "5.2.0",
    "@strapi/strapi": "5.2.0",
    "better-sqlite3": "11.3.0",
    "pg": "^8.13.1",
    "pg-connection-string": "^2.7.0",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "react-router-dom": "^6.0.0",
    "strapi-provider-upload-do": "^3.6.9",
    "styled-components": "^6.0.0"
  },

**Comment by reslear** [comment]
![Image](https://github.com/user-attachments/assets/e8b6a559-05be-497e-8cfb-6e8b3e4fbffd)

**Comment by Moonlight63** [comment]
I to am having the same issue. Glad I am not the only one as I thought it was something to do with my proxy config. Applying a patch to @strapi/admin to not use lodash set in 5.7.0 fixes it. I don't know if there is any other side effects from it though. Not an ideal solution. I have no custom local plugins and this is on a fresh install just for testing this exact issue. I did replicate some of my other production config by adding s3 provider, import-export, and advanced-uuid plugins through npm. I can test with a super fresh install, but really it doesn't matter because plugins are pretty much necessary in an actual production use.

**Comment by Yordan-Ramchev** [comment]
Same issue for multiple projects on separate different hosting providers:
- Railway
- Digital Ocean
- Hetzner
It's only reproduced on deployed environments (both npm v20 & yarn v4).

**Comment by blueagler** [comment]
> I to am having the same issue. Glad I am not the only one as I thought it was something to do with my proxy config. Applying a patch to @strapi/admin to not use lodash set in 5.7.0 fixes it. I don't know if there is any other side effects from it though. Not an ideal solution. I have no custom local plugins and this is on a fresh install just for testing this exact issue. I did replicate some of my other production config by adding s3 provider, import-export, and advanced-uuid plugins through npm. I can test with a super fresh install, but really it doesn't matter because plugins are pretty much necessary in an actual production use. I will also try upgrading to 5.8.0.
> 
> EDIT: After more testing, removing telemetryDisabled: true or setting it to false may or may not have had an impact. In my very unscientific test, I went from getting the loading screen 95% of the time to getting the loading screen 75% of the time. Applying a patch to @strapi/admin completely fixes it. There must be something wrong with the way lodash is working. :/
> 
> Updating to latest release didn't fix it. Here is my patch for 5.8.0. I don't know if you actually need to patch both the js and mjs file, but better safe than sorry I guess.
> 
> diff --git a/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-BjJpEjNZ.js b/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-BjJpEjNZ.js
> index 389b03c..98bcd89 100644
> --- a/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-BjJpEjNZ.js
> +++ b/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-BjJpEjNZ.js
> @@ -19,7 +19,6 @@ const Ornaments = require('./Ornaments-CID1aaLv.js');
>  const users = require('./users-DaPfjlwf.js');
>  const index = require('./index-DwA5cbf6.js');
>  const immer = require('immer');
> -const set = require('lodash/set');
>  const PrivateRoute = require('./PrivateRoute-qahg_uiu.js');
>  const cloneDeep = require('lodash/cloneDeep');
>  const useOnce = require('./useOnce-C7EQufL3.js');
> @@ -49,7 +48,6 @@ const React__namespace = /*#__PURE__*/_interopNamespace(React);
>  const lt__default = /*#__PURE__*/_interopDefault(lt);
>  const valid__default = /*#__PURE__*/_interopDefault(valid);
>  const get__default = /*#__PURE__*/_interopDefault(get);
> -const set__default = /*#__PURE__*/_interopDefault(set);
>  const cloneDeep__default = /*#__PURE__*/_interopDefault(cloneDeep);
>  
>  const name = "@strapi/admin";
> @@ -984,8 +982,11 @@ const initialState = {
>  const reducer = (state = initialState, action) => immer.produce(state, (draftState) => {
>    switch (action.type) {
>      case "SET_PLUGIN_READY": {
> -      set__default.default(draftState, ["plugins", action.pluginId, "isReady"], true);
> -      break;
> +      const newState = {
> +        ...state
> +      }
> +      newState.plugins[action.pluginId].isReady = true;
> +      return newState;
>      }
>      default:
>        return draftState;
> diff --git a/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-DsYL-HcK.mjs b/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-DsYL-HcK.mjs
> index 1d70944..1ba2b02 100644
> --- a/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-DsYL-HcK.mjs
> +++ b/node_modules/@strapi/admin/dist/admin/AuthenticatedLayout-DsYL-HcK.mjs
> @@ -15,7 +15,7 @@ import { L as LAYOUT_DATA, V as VerticalDivider, S as STATES, N as Number$1 } fr
>  import { g as getDisplayName, h as hashAdminUserEmail } from './users-8N93LH7R.mjs';
>  import { N as NpsSurvey } from './index-D4QvcqXm.mjs';
>  import { produce } from 'immer';
> -import set from 'lodash/set';
> +// import set from 'lodash/set';
>  import { P as PrivateRoute } from './PrivateRoute-DIxYvSni.mjs';
>  import cloneDeep from 'lodash/cloneDeep';
>  import { u as useOnce } from './useOnce-NHeEacbN.mjs';
> @@ -953,8 +953,11 @@ const initialState = {
>  const reducer = (state = initialState, action) => produce(state, (draftState) => {
>    switch (action.type) {
>      case "SET_PLUGIN_READY": {
> -      set(draftState, ["plugins", action.pluginId, "isReady"], true);
> -      break;
> +      const newState = {
> +        ...state
> +      }
> +      newState.plugins[action.pluginId].isReady = true;
> +      return newState;
>      }
>      default:
>        return draftState;
> For anyone that happens to come across this and doesn't know how to apply a patch in an npm project: npm install patch-package mkdir patches touch [@strapi](https://github.com/strapi)+admin+5.8.0.patch (copy above into that new file) in your package.json
> 
> "description": "A Strapi application",
>   "scripts": {
>     "build": "strapi build",
>     "deploy": "strapi deploy",
>     "develop": "strapi develop",
>     "start": "strapi start",
>     "strapi": "strapi",
>     "postinstall": "patch-package"    // <----- add this
>   },
>   "dependencies": {
> Now when you run npm install, this will apply the patch to the package. When building a docker container, remember to include the patches folder in the app root of the container along side the package.json file before you run npm install && npm build && npm prune --production
> 
> This was the only thing that works for me.

Patching with this will make graphql request throw errors.
[2025-01-25 01:36:52.308] http: POST /graphql (10 ms) 500
[2025-01-25 01:36:53.933] error: Cannot read properties of null (reading 'split')
TypeError: Cannot read properties of null (reading 'split')
    at splitDateString (node_modules/date-fns/parseISO.cjs:105:28)
    at Object.parseISO (node_modules/date-fns/parseISO.cjs:49:23)
    at Object.authenticate$2 [as authenticate] (node_modules/@strapi/admin/dist/server/index.js:545:77)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async Object.authenticate (node_modules/@strapi/core/dist/services/auth/index.js:50:24)
    at async bodyParser (node_modules/koa-bodyparser/index.js:78:5)
    at async cors (node_modules/@koa/cors/index.js:109:16)
    at async returnBodyMiddleware (node_modules/@strapi/core/dist/services/server/compose-endpoint.js:45:18)
    at async policiesMiddleware (node_modules/@strapi/core/dist/services/server/policy.js:15:5)
    at async node_modules/@strapi/core/dist/services/server/compose-endpoint.js:27:12

**Comment by markkaylor** [comment]
Hello I was able to reproduce the issue. I published an experimental to test the fix suggested in https://github.com/strapi/strapi/pull/22402 by @pksr . Could any of you try using it to confirm it fixes your issue in production?

```
  "dependencies": {
// ...
    "@strapi/plugin-cloud": "0.0.0-experimental.fd51913e4ee4e81a845e6f5bcfd4916cac70ca2e",
    "@strapi/plugin-seo": "^2.0.4",
    "@strapi/plugin-users-permissions": "0.0.0-experimental.fd51913e4ee4e81a845e6f5bcfd4916cac70ca2e",
    "@strapi/strapi": "0.0.0-experimental.fd51913e4ee4e81a845e6f5bcfd4916cac70ca2e",
// ...
```

**Comment by markkaylor** [comment]
Hello there's a new PR up: #22763 with a fix. I've tested with Strapi cloud, If @cslegany-synerinsoft or anyone else could test this experimental release: `0.0.0-experimental.a4df7ad5ff6a4366442059f7c067a0424ba1e2f9` to confirm on other providers that would be very helpful 🙏 

```
  "dependencies": {
// ...
    "@strapi/plugin-cloud": "0.0.0-experimental.a4df7ad5ff6a4366442059f7c067a0424ba1e2f9",
    "@strapi/plugin-seo": "^2.0.4",
    "@strapi/plugin-users-permissions": "0.0.0-experimental.a4df7ad5ff6a4366442059f7c067a0424ba1e2f9",
    "@strapi/strapi": "0.0.0-experimental.a4df7ad5ff6a4366442059f7c067a0424ba1e2f9",
// ...
```

**Comment by johnrobertmcc** [comment]
We also had trouble with Strapi stuck in loading state after authentication in production mode after upgrading to Strapi 5.8.x from Strapi 4 latest. We also were able to replicate it locally using `npm run build && npm run start` with both `NODE_ENV=production` and `NODE_ENV=local`. 

From what I can gather from the source code, there is a lifecycle issue in which Strapi admin does not finish waiting for the plugin to be `isReady` truthy, so it never triggers out of this block: https://github.com/strapi/strapi/blob/a4df7ad5ff6a4366442059f7c067a0424ba1e2f9/packages/core/admin/admin/src/components/PluginsInitializer.tsx#L49

I see that PR above is available, but we will wait for a stable version. 

Our temporary solution is to remove all offending plugins. Debugging steps for finding the plugin that's breaking for you:
- Open inspector while stuck in loading state
- Go to sources
- Open `AuthenticatedLayout__<hash>.js`
- Look for `isReady`, place debugger on that line
- Find which plugin is initialized on `isReady: false` (screenshot)

We were able to work around this plugin for now. If your offending plugin is a custom plugin, then the original solution of `isReady: true` on the `registerPlugin` should work. Otherwise, try removing the plugin if possible.

<img width="1233" alt="Image" src="https://github.com/user-attachments/assets/8accce1e-a3ee-44da-a9ef-def5342b9260" />


We also originally tried patching the package itself as suggested above, but this was the cleanest solution for us. Patching the package also doesn't guarantee that it moved out of that returns block.

**Comment by markkaylor** [comment]
Thank you for confirming @pksr

**Comment by veste-software** [comment]
Has this fix already a release on NPM?

**Comment by markkaylor** [comment]
> Has this fix already a release on NPM?

The fix will be released tomorrow in `5.9.1`

**Comment by thematrixdev** [comment]
> > Has this fix already a release on NPM?
> 
> The fix will be released tomorrow in `5.9.1`

I can find 5.9.0. Was 5.9.1 released?

**Comment by markkaylor** [comment]
The `5.9.1` release was delayed. It should happen this week. Usually we release every Wednesday.

**Comment by thematrixdev** [comment]
> > > > Has this fix already a release on NPM?
> > > 
> > > 
> > > The fix will be released tomorrow in `5.9.1`
> > 
> > 
> > I can find 5.9.0. Was 5.9.1 released?
> 
> The `5.9.1` release was delayed. It should happen this week. Usually we release every Wednesday.

The admin panel is inaccessible and unusable. Do release the patch as soon as possible please. 😭

![Image](https://github.com/user-attachments/assets/a7648c0a-f477-4bb2-b40b-836c425622cd)

**Comment by jorrit** [comment]
> The fix was just released in [5.10.0](https://github.com/strapi/strapi/releases/tag/v5.10.0)

Perhaps the release notes should contain a more explicit mention that this issue was fixed. Now it is just mentioned as `class Plugin is not immerable`.

**Comment by cacooper16** [comment]
Upgraded Strapi and yet to have the white screen issue. Seems to be moving quicker page-to-page too.

Thanks for your hard work. Great to have this fixed.

**Comment by obafunmiso-olufemi** [comment]
issue persist for me in 5.10.0.. no local plugin installed

**Comment by obafunmiso-olufemi** [comment]
@cacooper16 did you test this in deployed environment?

**Comment by cacooper16** [comment]
Hi @obafunmiso-olufemi, yes - this works for me on local and on deployed production. 

I am using Digital Ocean's App Platform.

Hope that helps.

**Comment by markkaylor** [comment]
> issue persist for me in 5.10.0 - 5.12.4 .. no local plugin installed And for everyone that says its working, did you test on cloud environment? Because it works fine locally for me but not working when I deployed [@markkaylor](https://github.com/markkaylor)

Do you have any plugins installed (not just local plugins)? Have you tried in an incognito browser? Do you have any errors in the console?

**Comment by aksbuzz** [comment]
e
